cpp/include/cudf/utilities/span.hpp

```diff
@@ -386,62 +409,44 @@ class base_2dspan {
 
   constexpr base_2dspan() noexcept = default;
   /**
-   * @brief Constructor a 2D span
+   * @brief Constructor from a span and number of elements in each row.
    *
-   * @param data Pointer to the data
-   * @param rows Number of rows
+   * @param flat_view The flattened 2D span
    * @param columns Number of columns
    */
-  constexpr base_2dspan(T* data, size_t rows, size_t columns) noexcept
-    : _data{data}, _size{rows, columns}
+  constexpr base_2dspan(RowType<T, dynamic_extent> flat_view, size_t columns) noexcept
+    : _flat{flat_view}, _size{flat_view.size() / columns, columns}
   {
   }
-  /**
-   * @brief Constructor a 2D span
-   *
-   * @param data Pointer to the data
-   * @param size Size of the 2D span as pair
-   */
-  base_2dspan(T* data, size_type size) noexcept : _data{data}, _size{std::move(size)} {}
 
   /**
    * @brief Returns a pointer to the beginning of the sequence.
    *
    * @return A pointer to the first element of the span
    */
-  constexpr auto data() const noexcept { return _data; }
+  constexpr auto data() const noexcept { return _flat.data(); }
+
   /**
    * @brief Returns the size in the span as pair.
    *
    * @return pair representing rows and columns size of the span
    */
   constexpr auto size() const noexcept { return _size; }
+
   /**
    * @brief Returns the number of elements in the span.
    *
    * @return Number of elements in the span
    */
-  constexpr auto count() const noexcept { return size().first * size().second; }
+  constexpr auto count() const noexcept { return _flat.size(); }
+
   /**
    * @brief Checks if the span is empty.
    *
    * @return True if the span is empty, false otherwise
    */
   [[nodiscard]] constexpr bool is_empty() const noexcept { return count() == 0; }
 
-  /**
-   * @brief Returns flattened index of the element at the specified 2D position.
-   *
-   * @param row The row index
-   * @param column The column index
-   * @param size The size of the 2D span as pair
-   * @return The flattened index of the element at the specified 2D position
-   */
-  static constexpr size_t flatten_index(size_t row, size_t column, size_type size) noexcept
```

Comment:
Removed since it's used in a very limited way. The 2D span can really only be row-major with the current API, so we don't need to expose the orientation through this (I think that was the original intent).